csrc/ops/composite.cpp

```diff
@@ -74,6 +81,13 @@ TensorView* matmul(TensorView* a, TensorView* b) {
   return fusedMultiplySum(tv0b, tv1b, {-2});
 }
 
+} // namespace
+
+TensorView* matmul(TensorView* a, TensorView* b) {
+  auto* tv = matmulImpl(a, b);
+  return castOp(a->getDataType().value(), tv);
```

Comment:
nitpick: it reads cleaner to me to have the
```
  NVF_CHECK(
      a->getDataType().value() == b->getDataType().value(),
      "data types of inputs to matmul don't match");
```

before casting output to one of the input types.

So can we move this cast inside `matmulImpl`?

Reply:
Thanks! Please let me know if I was not able to address your comment.